dietpi-letsencrypt: update HTTP/2 syntax and add HTTP/3/QUIC

JappeHallunken opened the pull request:
Nginx throws a warning at startup, because

> "listen ... http2" directive is deprecated, use the "http2" directive instead

Changed from
```
listen [::]:443 ssl http2 ipv6only=on; # managed by Certbot
listen 443 ssl http2; # managed by Certbot
```
to
```
listen [::]:443 ssl ipv6only=on; # managed by Certbot
listen 443 ssl; # managed by Certbot
http2 on;
```

JappeHallunken committed:
correct syntax for enabling http2 in nginx. current syntax is deprecated

MichaIng reviewed (changes requested):
This directive appeared with Nginx 1.25.1, hence is not available on Debian Bookworm and below yet. On Trixie, we might want to add the `quic` parameter instead. Weirdly there is a `http3` directive, which is however enabled by default, but without QUIC it seems, and the `http2` directive is still off by default. So inconsistent. So:
- Up to Bookworm: Keep things as they are
- From Trixie on: Replace `http2` argument with `quic`, and add `http2` directive instead.

JappeHallunken committed:
Enhance Nginx config for HTTP/2 and QUIC support

Updated Nginx configuration to support HTTP/2 and QUIC.

JappeHallunken committed:
Enhance Nginx config for HTTP/2 and QUIC

remove newline to pass shellcheck

MichaIng commented on dietpi/dietpi-letsencrypt:
```diff
@@ -225,7 +225,8 @@ _EOF_
 			fi
 
 			# Enable HTTP/2: https://github.com/certbot/certbot/issues/3646
-			grep -q 'http2' /etc/nginx/sites-available/default || G_EXEC sed --follow-symlinks -i '/^[[:blank:]]*listen /s/443 ssl/443 ssl http2/' /etc/nginx/sites-available/default
```
I guess you meant to keep this line, and then update it below on Debian Trixie and above, or if `dietpi-letsencrypt` is re-applied, right?

Or does Certbot add the `http2` parameter in the meantime? I'll try to add the individual lines one by one, preserving any existing directive, else it seems a little fragile for duplicate directives and possible breakage on custom `default` configs.

MichaIng committed:
dietpi-letsencrypt: preserve Bookworm support

Apply "http2" parameter on Bookworm, and new "http2" directive as well as HTTP/3 with QUIC from Trixie on, where it is supported.

Also apply each directive separately, always preserving any existing matching directive.
